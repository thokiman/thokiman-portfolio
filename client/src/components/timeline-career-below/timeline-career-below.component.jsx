import React from "react";
import { GiMechaMask, GiTribalMask } from "react-icons/gi";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import {
  toggleYear2017Click,
  toggleYear2019Click,
} from "../../redux/about/about.actions";
import {
  selectIsYear2017Click,
  selectIsYear2019Click,
} from "../../redux/about/about.selectors";
import TimelineCareerLink from "../timeline-career-link/timeline-career-link.component";
import { BelowTimelineHeaderContainer } from "./timeline-career-below.styles";

export const TimelineCareerBelow = ({
  isYear2017Click,
  isYear2019Click,
  toggleYear2017Click,
  toggleYear2019Click,
}) => {
  return (
    <BelowTimelineHeaderContainer>
      <TimelineCareerLink
        isClick={isYear2017Click}
        yearClick={toggleYear2017Click}
        TimelineIcon={GiTribalMask}
        placement="below"
        year="2017"
      />
      <TimelineCareerLink
        isClick={isYear2019Click}
        yearClick={toggleYear2019Click}
        TimelineIcon={GiMechaMask}
        placement="below"
        year="2019"
      />
    </BelowTimelineHeaderContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isYear2017Click: selectIsYear2017Click,
  isYear2019Click: selectIsYear2019Click,
});

const mapDispatchToProps = (dispatch) => ({
  toggleYear2017Click: () => dispatch(toggleYear2017Click()),
  toggleYear2019Click: () => dispatch(toggleYear2019Click()),
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TimelineCareerBelow);
