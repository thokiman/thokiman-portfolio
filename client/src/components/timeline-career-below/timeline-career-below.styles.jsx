import styled from "styled-components";

export const BelowTimelineHeaderContainer = styled.div`
  grid-area: below-timeline-header-container;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr;
  margin: 20px 0;
  cursor: pointer;
  height: 100%;
`;

BelowTimelineHeaderContainer.displayName = "BelowTimelineHeaderContainer";
