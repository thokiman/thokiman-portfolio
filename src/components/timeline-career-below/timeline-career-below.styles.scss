  .below-timeline-header-container {
      grid-area: below-timeline-header-container;
      display: grid;
      grid-template-columns: repeat(3, 1fr);
      grid-template-rows: 1fr;

      margin: 20px 0;
      cursor: pointer;
      height: 100%;



  }