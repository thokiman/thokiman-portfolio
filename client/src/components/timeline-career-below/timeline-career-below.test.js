import React from "react";
import { shallow } from "enzyme";

import { TimelineCareerBelow } from "./timeline-career-below.component";

describe("TimelineCareerBelow component", () => {
  let wrapper;
  let mockToggleYear2017Click;
  let mockToggleYear2019Click;
  const mockFalseBool = false;
  beforeEach(() => {
    mockToggleYear2017Click = jest.fn();
    mockToggleYear2019Click = jest.fn();
    const mockProps = {
      isYear2017Click: mockFalseBool,
      isYear2019Click: mockFalseBool,
      toggleYear2017Click: mockToggleYear2017Click,
      toggleYear2019Click: mockToggleYear2019Click,
    };
    wrapper = shallow(<TimelineCareerBelow {...mockProps} />);
  });

  it("should render TimelineCareerBelow component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
