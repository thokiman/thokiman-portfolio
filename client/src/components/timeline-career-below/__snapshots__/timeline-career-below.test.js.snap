// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineCareerBelow component should render TimelineCareerBelow component 1`] = `ShallowWrapper {}`;
